SAP/spartacus
# test(ssr-tests): print SSR Logs for errors in SSR tests, ALSO when a test times out

Previously, timed out tests didn't get the SSR logs printed. Now it's fixed.

To make it happen, we're changing an approach. Now, instead of wrapping each test callback of `it()`, we're setting up a global Jest Environment class, which intercepts the events of failed tests and attaches SSR logs to those error objects in a generic way.

**QA steps**
1. Prerequisites
   1. Deliberately make one test time out - e.g. add very short timeout `1_000` (1 second) timeout to the first test
   1. Deliberately make one of the tests fail - e.g. change the expected HTTP status from `200` to `123` to the second test
   
   See the example below:
   ![image](https://github.com/user-attachments/assets/10147a28-a802-4622-a10e-e77aa789f9e5)


1. Build libs, app and SSR (with localhost backend):
   `npm run build:libs && npm run build && build:ssr:local-http-backend`
3. Run ssr-tests:
   `npm run test:ssr`
4. Verify that BOTH Error messages printed by JestJS contain SSR logs as a context (both the test that timed our and the test that had a failed `expect()`):

BEFORE: (screenshot just for the timed out test)

![image](https://github.com/user-attachments/assets/f17a6c3e-9e31-47d4-9657-1665ae9c3c0a)

AFTER: (screenshot for both the timed out test and for the failed expect test)

![image](https://github.com/user-attachments/assets/dcef2e1d-6c2c-4f60-b085-a9112a1e547b)
![image](https://github.com/user-attachments/assets/7fb07f72-0fd4-48d2-a107-ea9bcda3148a)
![image](https://github.com/user-attachments/assets/c047e1d6-2af9-4667-9ba0-e598e4a23c1f)
(...)

It's a followup fix for https://jira.tools.sap/browse/CXSPA-8564
